rust-server/src/main.rs

```diff
@@ -105,21 +141,28 @@ async fn main() -> Result<()> {
     let http_server = axum::serve(listener, app)
         .into_future()
         .map_err(anyhow::Error::from);
-    let term_sigs = pin!(handle_main_signals()?);
+    let term_sigs = pin!(handle_main_signals(onexit)?);
     let either = future::select(http_server, term_sigs).await;
     either.factor_first().0
 }
 
-fn handle_main_signals() -> Result<impl Future<Output = Result<()>>> {
+fn handle_main_signals<F, Fut>(onexit: F) -> Result<impl Future<Output = Result<()>>>
+where
+    F: FnOnce() -> Fut,
+    Fut: Future<Output = ()>,
+{
     // On Windows we probably wouldn't have this one. Also what about the fact this is normally for
     // user detected errors?
+    let interrupt = signal("SIGINT", SignalKind::interrupt())?;
```

Comment:
As mentioned, I was trying to use sigint to force flushing which applies to the JsonFiles backend. This isn't conventional and there's probably better ways to do it (maybe a http endpoint to trigger flush). Maybe as a compromise I only hook sigint for this purpose for the JsonFiles backend and leave it as people expect otherwise?